tensorflow/core/kernels/data/captured_function.h

```diff
@@ -211,16 +211,21 @@ class InstantiatedCapturedFunction {
   // Runs the instantiated captured function. This method takes ownership of
   // the tensors in `args`, in order to be able to deallocate them as early as
   // possible. Use `RunWithBorrowedArgs()` if the caller needs to retain
-  // ownership of the `args`.
+  // ownership of the `args`. Pass non-null `node` to record processing time
+  // for modeling Iterator's GetNext() resource usage.
   Status Run(IteratorContext* ctx, std::vector<Tensor>&& args,
-             std::vector<Tensor>* rets) const;
+             std::vector<Tensor>* rets,
+             const std::shared_ptr<model::Node>& node = nullptr) const;
```

Comment:
Please do not use default arguments as they make it difficult to extend the API. Instead add the overload as per my suggestion.